gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1065,13 +1065,14 @@ public void run() {
             callbacks.getStoragePoolSize(true);
       }
       this.memorySize = memorySize;
-      if (memorySize > 0) {
+      if (memorySize > 0 && CallbackFactoryProvider.getStoreCallbacks().isEnterpriseEdition()) {
```

Comment:
Why through callback? Why not just check GemFireVersion.isEnterpriseEdition? That would have been loaded appropriately by now so that check works fine -- at least worked well in my local changes.